parts/k8s/addons/cloud-node-manager.yaml

```diff
@@ -88,18 +88,52 @@ spec:
         command:
         - cloud-node-manager
         - --node-name=$(NODE_NAME)
+        {{- if IsAzureStackCloud}}
+        - --use-instance-metadata=false
+        - --cloud-config=/etc/kubernetes/azure.json
```

Comment:
do you know why only ASH needs `cloud-config` and `kubeconfig`?

Reply:
For `kubeconfig`, if not provided the ecp (external cloud provider) will use default value from environment variable of `Kubernetes Service` (https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/vendor/k8s.io/client-go/rest/config.go#L488), which doesn't work for Azure Stack as well as Azure Windows node.

For `cloud-config`, we need that to correct get cloud and tenant information.

Reply:
Ideally, we should figure which `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` values work on ASH and Windows. The same for cloud and tenant info. The least access to the host file system, the better.